File: CSharpProjects/CSharpProjects/BubbleSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpProjects
{
    internal class BubbleSort
    {
        static void Main(string[] args)
        {
            int[] arr = { 5, 4, 3, 2, 1 };
            for(int i = arr.Length - 1; i >= 1; i--)
            {
                for(int j = 0;j<=i - 1; j++)
                {
                    if (arr[j] > arr[j+1]) Swap(arr, j,j+1);
                }
            }

            foreach(int ele in arr) Console.Write(ele + " ");
        }

        public static void Swap(int[] arr,int i ,int j)
        {
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }


    }
}

File: JAVA DSA/BubbleSort.java
public class BubbleSort {
    public static void main(String[] args) {
        int [] arr = {1,2,3,4,5,6};
        
        for(int i=arr.length-1; i>=1;i--){
            int didswap=0;
            for(int j=0;j<=i-1;j++){
                if(arr[j] > arr[j+1]){
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]= temp;
                    didswap = 1;
                }
            }
            if(didswap == 0) break;     //optimised for best case, in case all sorted
        }
        for(int ele:arr)System.out.print(ele + " ");

    }
}
